File: Algorithms and Data Structures/Algorithms/Sorting Algorithms/bubble_sort.cpp
#include <iostream>
#include <ctime>

using namespace std;

void swap(int &x, int &y){
    int temp = x;
    x = y;
    y = temp;
}

void bubbleSort(int arr[], int size){
    for(int i = 0; i < size; ++i){
        for(int j = i + 1; j < size; ++j){

            //ascending - <
            //descending - >

            if(arr[j] < arr[i]){
                swap(arr[i], arr[j]);
            }
        }
    } 
}

int main(){
    srand(time(NULL));

    cout << "Enter size of an array: " << endl;
    int size;
    cin >> size;
    int arr[size];

    for (size_t i = 0; i < size; ++i){
        arr[i] = rand() % size;
    }

    cout << "unsorted array:" << endl;

    for (size_t i = 0; i < size; ++i){
        cout << arr[i] << " ";
    }
    cout << endl;

    cout << "sorted array:" << endl;

    bubbleSort(arr, size);

    for (int i = 0; i < size; ++i){
        cout << arr[i] << " ";
    }
    return 0;
}

File: Algorithms and Data Structures/Algorithms/Sorting Algorithms/bubble_sort.py
def bubble_sort(arr):
    for i in range(len(arr)-1):
        for j in range(0, len(arr)-i-1):
            if arr[j] > arr[j + 1] :
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

numbers = list(map(int, input().split()))

bubble_sort(numbers)

for i in numbers:
    print(i, end=' ')
    